feat: create `no-interpolation-in-snapshots` rule (#553)
jest-community/eslint-plugin-jest: docs/rules/no-interpolation-inline-snapshot.md

> # No string interpolation inside inline snapshots (no-interpolation-inline-snapshot)

Reviewer: ```suggestion
# Disallow string interpolation inside inline snapshots (`no-interpolation-inline-snapshot`)
```

Prevents the use of string interpolations within `toMatchInlineSnapshot`.

## Rule Details

> Interpolation prevents snapshots from being updated. Instead, properties should
> be overloaded with a matcher by passing the optional `propertyMatchers` argument
> to `toMatchInlineSnapshot`.

Reviewer: ```suggestion
Interpolation prevents snapshots from being updated. Instead, properties should
be overloaded with a matcher by using [property matchers](https://jestjs.io/docs/en/snapshot-testing#property-matchers).
```

> Examples of **incorrect** code for this rule:

Reviewer: Could you chuck a in/correct example for `toThrowErrorMatchingInlineSnapshot` as well?

Author: added in https://github.com/jest-community/eslint-plugin-jest/pull/553/commits/a405636f7128de898187ac9dd3b54e90d65529b2
